src/pending.js

```diff
@@ -0,0 +1,128 @@
+var _ = require('lodash');
+var debug = require('debug')('stats-collector:pending');
+var taskcluster = require('taskcluster-client');
+
+const FLUSH_PERIOD = 30; // seconds
+
+// check for pending tasks that aren't really pending with this frequency; should
+// be smaller than MIN_CHECK_AGE
+const CHECK_INTERVAL = 120; // seconds
+
+// only check tasks pending at least this long; this should be lower than any alert
+// thresholds for pending tasks.
+const MIN_CHECK_AGE = 300; // seconds
+
+// calculate the taskKey and scheduled date with the earliest scheduled time for
+// the given workerType; if there are none, returns null and the current time
+var earliest = (state, workerType, now) => {
+  return _.reduce(state.pendingTasks[workerType], (res, scheduled, taskKey) => {
```

Comment:
I think you want `_.maxBy(state.pendingTasks[workerType], 'scheduled')` :)

You can also pass a function... instead of just a key..
Personally, I'm no fan of reduce, it seems to be a lot harder than simple iteration... but _.maxBy is always better :)